### PR TITLE
feat(operators): Use lift in the operators that don't currently use lift.

### PR DESCRIPTION
**Pull request by trxcllnt**
**Description:**
Use lift in the operators that don't currently use lift.

**Related issue (if exists):**
#1829

### CODE REVIEW

File: src/operator/merge.ts [hunk 1]
```diff
@@ -149,7 +148,7 @@ export function mergeStatic<T, R>(...observables: Array<ObservableInput<any> | S
     concurrent = <number>observables.pop();
   }
 
-  if (observables.length === 1) {
+  if (scheduler === null && observables.length === 1) {
```
Comment (jayphelps) [inline src/operator/merge.ts]: did this fix an unreported bug?

File: src/Rx.ts [hunk 2]
```diff
@@ -150,7 +150,6 @@ export {Subscriber} from './Subscriber';
 export {AsyncSubject} from './AsyncSubject';
 export {ReplaySubject} from './ReplaySubject';
 export {BehaviorSubject} from './BehaviorSubject';
-export {MulticastObservable} from './observable/MulticastObservable';
```
Comment (jayphelps) [inline src/Rx.ts]: This will require a BREAKING CHANGE note, if accepted.

File: src/operator/race.ts [hunk 3]
```diff
@@ -23,8 +23,7 @@ export function race<T>(...observables: Array<Observable<T> | Array<Observable<T
     observables = <Array<Observable<T>>>observables[0];
   }
 
-  observables.unshift(this);
-  return raceStatic.apply(this, observables);
+  return this.lift.call(raceStatic<T>(this, ...observables));
```
Comment (jayphelps) [inline src/operator/race.ts]: Without `observables.unshift(this)` doesn't this now have different behavior ? Was this a bug?

File: src/operator/merge.ts (hunk 1 above)
Comment (trxcllnt) [inline src/operator/merge.ts]: oh weird, that must have slipped in when I branched from master. yes, it's a bug fix that I was tracking down earlier today but didn't get to finish writing a test for. I can try to remove it from this change set, or alternatively add the test I was going to write and push again :[

File: src/Rx.ts (hunk 2 above)
Comment (trxcllnt) [inline src/Rx.ts]: bummertown. I'll amend the commit message.

**Comment by jayphelps** [comment]
Maybe I'm missing something but doesn't this break static operator usage, which leads me to believe we're _not currently testing them like that_ 😱😱😱.

e.g. [static `merge`](https://github.com/trxcllnt/RxJS/blob/241787ed5b415f4ef6613b2c9de4dade89d7ae8c/src/observable/merge.ts) just references the instance operator function.

```js
import { merge } from 'rxjs/observable/merge';

Observable.merge(stream1, stream2);
// or
merge(stream1, stream2);
```

As well as
```js
import { Observable } from 'rxjs/Observable';
import 'rxjs/add/observable/merge';

Observable.merge(stream1, stream2);
```

Did you have thoughts on the somewhat related issue of static methods not using lift?

https://github.com/ReactiveX/rxjs/issues/1829#issuecomment-239278815

**Comment by jayphelps** [comment]
I totally missed that `Observable.merge` aka static merge does not use the regular merge, instead using `mergeStatic` which doesn't rely on `this`. So perhaps I'm drunk and this isn't an issue. Though I still would like static methods to lift too, but we could in fact punk on that regarding this PR.

File: src/operator/race.ts (hunk 3 above)
Comment (trxcllnt) [inline src/operator/race.ts]: Shouldn't be. The instance method unwraps the argument if it's an Array of Observables, but I just looked and we don't have any test for this, which makes me nervous.

Comment (jayphelps) [inline src/operator/race.ts]: image

**Comment by trxcllnt** [comment]
@jayphelps I'm also keen to update the static methods to use a creation/conversion method, but I wanted to get the instance methods out of the way first, as I imagine it'll be a bit before the static method is finalized.

File: src/operator/concat.ts [hunk 4]
```diff
@@ -119,5 +119,9 @@ export function concatStatic<T, R>(...observables: Array<ObservableInput<any> |
     scheduler = args.pop();
   }
 
+  if (scheduler === null && observables.length === 1) {
+    return <Observable<R>>observables[0];
+  }
+
```
Comment (jayphelps) [inline src/operator/concat.ts]: is this another one that slipped in or?

Comment (trxcllnt) [inline src/operator/concat.ts]: heh yep

Comment (trxcllnt) [inline src/operator/concat.ts]: added a test for this

File: src/operator/merge.ts (hunk 1 above)
Comment (trxcllnt) [inline src/operator/merge.ts]: added a test for this

File: src/operator/race.ts (hunk 3 above)
Comment (trxcllnt) [inline src/operator/race.ts]: added a test for this

**Comment by jayphelps** [comment]
LGTM, but needs rebase.

**Comment by jayphelps** [comment]
@trxcllnt I forgot to remind you: this needs a BREAKING CHANGE added to the commit for the removal of `MulticastObservable`.

**Comment by trxcllnt** [comment]
@jayphelps done

**Comment by jayphelps** [comment]
LGTM

**Pull request merged**